lighthouse-core/audits/script-treemap-data.js

```diff
@@ -0,0 +1,262 @@
+        if (data.unusedBytes) node.unusedBytes = (node.unusedBytes || 0) + data.unusedBytes;
+
+        // Only leaf nodes might have duplication data.
+        if (isLastSegment && data.duplicatedNormalizedModuleName !== undefined) {
+          node.duplicatedNormalizedModuleName = data.duplicatedNormalizedModuleName;
+        }
+      });
+    }
+
+    // For every source file, apply the data to all components
+    // of the source path, creating nodes as necessary.
+    for (const [source, data] of Object.entries(sourcesData)) {
+      addAllNodesInSourcePath(source || `<unmapped>`, data);
+    }
+
+    /**
+     * Collapse nodes that have only one child.
+     * @param {Node} node
+     */
+    function collapseAll(node) {
+      while (node.children && node.children.length === 1) {
+        node.name += '/' + node.children[0].name;
+        node.children = node.children[0].children;
+      }
+
+      if (node.children) {
+        for (const child of node.children) {
+          collapseAll(child);
+        }
+      }
+    }
+    collapseAll(topNode);
+
+    return topNode;
+  }
+
+  /**
+   * Returns root node containers where the first level of nodes are script URLs.
+   * If a script has a source map, that node will be set by prepareTreemapNodes.
+   * @param {LH.Artifacts} artifacts
+   * @param {LH.Audit.Context} context
+   * @return {Promise<RootNodeContainer[]>}
+   */
+  static async makeRootNodes(artifacts, context) {
+    /** @type {RootNodeContainer[]} */
+    const rootNodeContainers = [];
+
+    let inlineScriptLength = 0;
+    for (const scriptElement of artifacts.ScriptElements) {
+      // No src means script is inline.
+      // Combine these ScriptElements so that inline scripts show up as a single root node.
+      if (!scriptElement.src) {
+        inlineScriptLength += (scriptElement.content || '').length;
+      }
+    }
+    if (inlineScriptLength) {
+      const name = artifacts.URL.finalUrl;
+      rootNodeContainers.push({
+        name,
+        node: {
+          name,
+          resourceBytes: inlineScriptLength,
+        },
+      });
+    }
+
+    const bundles = await JsBundles.request(artifacts, context);
+    const duplicationByPath = await ModuleDuplication.request(artifacts, context);
+
+    for (const scriptElement of artifacts.ScriptElements) {
+      if (!scriptElement.src) continue;
+
+      const name = scriptElement.src;
+      const bundle = bundles.find(bundle => scriptElement.src === bundle.script.src);
+      const scriptCoverages = artifacts.JsUsage[scriptElement.src];
+      if (!bundle || !scriptCoverages) {
+        rootNodeContainers.push({
+          name,
+          node: {
+            name,
+            resourceBytes: scriptElement.src.length,
+          },
+        });
+        continue;
+      }
+
+      const unusedJavascriptSummary = await UnusedJavaScriptSummary.request(
+        {url: scriptElement.src, scriptCoverages, bundle}, context);
+
+      /** @type {Node} */
```

Comment:
nit: don't need this, tsc's got your back :)

Reply:
This can change what line an error is reported on, and makes the error message less complex. You can see the difference by commenting out the `name` property and seeing how the error changes.